go-controller/pkg/node/ovspinning/ovspinning_linux_test.go

```diff
@@ -0,0 +1,187 @@
+		assert.NoError(t, err)
+
+		klog.Infof("Test CPU Affinity %x", tmpCPUset)
+
+		assertPIDHasSchedAffinity(t, ovsVSwitchdPid, tmpCPUset)
+		assertPIDHasSchedAffinity(t, ovsDBPid, tmpCPUset)
+	}
+
+	// Disable the feature by making the enabler file empty
+	ioutil.WriteFile(featureEnablerFile, []byte(""), 0)
+	assert.NoError(t, err)
+
+	var tmpCPUset unix.CPUSet
+	tmpCPUset.Set(0)
+	err = unix.SchedSetaffinity(os.Getpid(), &tmpCPUset)
+	assert.NoError(t, err)
+
+	assertNeverPIDHasSchedAffinity(t, ovsVSwitchdPid, tmpCPUset)
+	assertNeverPIDHasSchedAffinity(t, ovsDBPid, tmpCPUset)
+}
+
+func TestIsFileNotEmpty(t *testing.T) {
+
+	defer mockFeatureEnableFile(t, "")()
+
+	result, err := isFileNotEmpty(featureEnablerFile)
+	assert.NoError(t, err)
+	assert.False(t, result)
+
+	ioutil.WriteFile(featureEnablerFile, []byte("1"), 0)
+	result, err = isFileNotEmpty(featureEnablerFile)
+	assert.NoError(t, err)
+	assert.True(t, result)
+
+	os.Remove(featureEnablerFile)
+	result, err = isFileNotEmpty(featureEnablerFile)
+	assert.NoError(t, err)
+	assert.False(t, result)
+}
+
+func mockOvsdbProcess(t *testing.T) (int, func()) {
+	ctx, stopCmd := context.WithCancel(context.Background())
+	defer stopCmd()
+
+	cmd := exec.CommandContext(ctx, "sleep", "10")
+
+	err := cmd.Start()
+	assert.NoError(t, err)
+
+	previousGetter := getOvsDBServerPIDFn
+	getOvsDBServerPIDFn = func() (string, error) {
+		return fmt.Sprintf("%d", cmd.Process.Pid), nil
+	}
+
+	return cmd.Process.Pid, func() {
+		stopCmd()
+		getOvsDBServerPIDFn = previousGetter
+	}
+}
+
+func mockOvsVSwitchdProcess(t *testing.T) (int, func()) {
+	ctx, stopCmd := context.WithCancel(context.Background())
+	defer stopCmd()
+
+	cmd := exec.CommandContext(ctx, "sleep", "10")
+
+	err := cmd.Start()
+	assert.NoError(t, err)
+
+	previousGetter := getOvsVSwitchdPIDFn
+	getOvsVSwitchdPIDFn = func() (string, error) {
+		return fmt.Sprintf("%d", cmd.Process.Pid), nil
+	}
+
+	return cmd.Process.Pid, func() {
+		stopCmd()
+		getOvsVSwitchdPIDFn = previousGetter
+	}
+}
+
+func setTickDuration(d time.Duration) func() {
+	previousValue := tickDuration
+	tickDuration = d
+
+	return func() {
+		tickDuration = previousValue
+	}
+}
+
+func mockFeatureEnableFile(t *testing.T, data string) func() {
+
+	f, err := ioutil.TempFile("", "enable_dynamic_cpu_affinity")
+	assert.NoError(t, err)
+
+	previousValue := featureEnablerFile
+	featureEnablerFile = f.Name()
+
+	ioutil.WriteFile(featureEnablerFile, []byte(data), 0)
+	assert.NoError(t, err)
+
+	return func() {
+		featureEnablerFile = previousValue
+		os.Remove(f.Name())
+	}
+}
+
+func assertPIDHasSchedAffinity(t *testing.T, pid int, expectedCPUSet unix.CPUSet) {
+	var actual unix.CPUSet
+	assert.Eventually(t, func() bool {
+		err := unix.SchedGetaffinity(pid, &actual)
+		assert.NoError(t, err)
+
+		return actual == expectedCPUSet
+	}, time.Second, 10*time.Millisecond, "pid[%d] Expected CPUSet %0x != Actual CPUSet %0x", pid, expectedCPUSet, actual)
```

Comment:
Is this `assert.Eventually` timing out after only 1 second? I'm asking because in `Run()` we're aligning ovs with ovnk cpuset every 10 seconds.

Reply:
No, the tick duration is changed during the test by 
```
	defer setTickDuration(20 * time.Millisecond)()
```

Keeping the tick to 10 seconds would make the test very slow.